### PR TITLE
docs: add hotdoc

### PR DESCRIPTION
**Pull request by ArsenArsen**

### CODE REVIEW

File: docs/.gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+build
+hotdoc-private-libasync
```
Comment (avdgrinten) [inline docs/.gitignore:2]: When is that used?

File: docs/meson.build [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+	sitemap: 'sitemap',
+	index: 'src/index.md',
+	extra_assets: [ 'assets/' ],
+)
+
+
+custom_target('libasync-doc-always',
+	build_by_default: true,
+	build_always_stale: true,
+	output: 'phony',
```
Comment (avdgrinten) [inline docs/meson.build:15]: I'd rather have a `run_target` without the `phony` hack (even if that means invoking it manually).

File: docs/.gitignore (hunk 1 above)
Comment (ArsenArsen) [reply docs/.gitignore:2]: it's not anymore, residue from testing. will remove

File: docs/meson.build [hunk 3]
```diff
@@ -0,0 +1,14 @@
+project('libasync_docs')
```
Comment (avdgrinten) [inline docs/meson.build:1]: Please merge the content of this file into the top-level `meson.build.`

**Comment by avdgrinten** [comment]
Looks good now, thank you!

**Pull request merged**